src/lib/realmdb/RealmDB.js

```diff
@@ -300,6 +348,68 @@ class RealmDB implements DB {
     log.debug('getFeedPage result:', numResults, offset, res.length, res)
     return res
   }
+
+  //TODO:  make sure profile contains walletaddress or enforce it in schema in realmdb
+  setProfile(profile) {
+    return this._profiles().updateOne(
+      { user_id: this.user.id },
+      { user_id: this.user.id, ...profile },
+      { upsert: true },
+    )
+  }
+
+  /**
+   * read the complete raw user profile from realmdb. result fields might be encrypted
+   *  @returns {Promise<any>}
+   */
+  getProfile(): Promise<any> {
+    return this._profiles().findOne({ user_id: this.user.id })
+  }
+
+  /**
+   * get user profile from realmdb. result fields might be encrypted
+   * @param key
+   * @param field
+   * @returns {Promise<any | null>}
+   */
+  getProfileByField(key: string, field: string): Promise<any> {
+    return this._profiles().findOne({ [`${key}.display`]: field })
+  }
+
+  /**
+   * get user profile from realmdb by WalletAddress. result fields might be encrypted
+   * @param walletAddress
+   * @returns {Promise<any | null>}
+   */
+  getProfileByWalletAddress(walletAddress: string): Promise<any> {
+    return this.getProfileByField('walletAddress', walletAddress)
+  }
+
+  /**
+   * Set profile fields
+   * @param fields
+   * @returns {Promise<Realm.Services.MongoDB.UpdateResult<any>>}
+   */
+  setProfileFields(fields: { key: String, field: ProfileField }): Promise<any> {
+    return this._profiles().updateOne({ user_id: this.user.id }, { $set: fields })
+  }
+
+  /**
+   * Removing the field from record
+   * @param field
+   * @returns {Promise<Realm.Services.MongoDB.UpdateResult<*>>}
+   */
+  removeField(field) {
+    return this._profiles().updateOne({ user_id: this.user.id }, { $unset: { [field]: { value: null } } })
```

Comment:
1. the value for unset isnt necessary
2. in userprofilestorage either use removefield or use setProfileFields({[key]: null} , decide which way you are going to do, and remove the other way code